src/lib/util/zippath.cpp

```diff
@@ -111,25 +111,25 @@ int is_path_separator(char c)
  * @param [in,out]  endpos      If non-null, the endpos.
  */
 
-static void parse_parent_path(const char *path, int *beginpos, int *endpos)
+static void parse_parent_path(const std::string &path, std::string::size_type *beginpos, std::string::size_type *endpos)
 {
-	int length = strlen(path);
-	int pos;
+	std::string::size_type length = path.length();
+	std::string::size_type pos;
 
-	/* skip over trailing path separators */
+	// skip over trailing path separators
 	pos = length - 1;
-	while((pos > 0) && is_path_separator(path[pos]))
+	while ((pos > 0) && is_path_separator(path[pos]))
```

Comment:
This needs to change to `((pos > 0) && (pos != std::string::npos) && is_path_separator(path[pos]))`